.github/dependabot.yml

```
@@ -0,0 +1,31 @@
+version: 2
+updates:
+- package-ecosystem: gomod
+  directory: /
+  schedule:
+    interval: weekly
+    day: wednesday
+    time: '08:00' # UTC
+  labels:
+  - priority/important-longterm
+  - area/dependency
+  - kind/cleanup
```

Bumping dependency may introduce feature, it won't clean anything in our codebase. 
```suggestion
  - kind/feature
```

---

How's a dependency bump a feature from the operator's user perspective? Remember this also propagates to the release notes. I think cleanup fits fine, as per the label's comment, it "Categorizes issue or PR as related to cleaning up code, process, or technical debt."

---

We use dependencies for bunch of stuff, If for example there's performance improvement in the dependent library, then Operator can gain this performance improvement as well if it's in its hot branch. Performance improvements are features. 
Or if Scylla Manager starts supporting new bucket location, then bumping manager client may relax validation unblocking new feature in Operator. 

We use dependencies for features they provide, so even a bug fix there is a feature for us. 

> I think cleanup fits fine, as per the label's comment, it "Categorizes issue or PR as related to cleaning up code, process, or technical debt."

I don't agree, dependency bump doesn't fall into any of these categories.

---

I feel about this in two different ways at once:

1. In the current system, I think that dependency bumps deserve their own category (say `kind/dependency-bump`) 
2. Usefulness of dep bumps in the release notes commit-per-commit is minimal, - what matters is:
    - Scylla ecosystem default version changes
    - Scylla ecosystem supported version range changes
    - non-Scylla-ecosystem co-installed software default versions (prom, grafana, etc)
    - and maybe a diff of go.mod, and (even less likely) base image updates.

My suggestion is to, in this order:

In scope of this PR:
1. introduce `kind/dependency-bump`
2. if we don't want to introduce `kind/dependency-bump`, keep `kind/cleanup` because version bumps IMO are noise to a user skimming through a list of functional changes

Outside of the scope of this PR: get the dep bumps excluded from the changelog and include a report of version changes

---

Replying directly to @zimnx's comment as we were responding simultaneously :). This is a paraphrase of the above, addressing the comment directly.

I agree that version bumps can bring useful value to the user. I however don't believe that (in a typical case) they're functional changes and therefore don't belong under `kind/feature`.

Therefore I suggest that we find a way to keep `kind/feature` reserved for functional changes and to present dependencies in a useful way.

(I think we may be hitting the limits of our changelog tooling)

---

I would be ok with new kind

---

Done - PTAL @zimnx 

Also adding the label to labelsync: https://github.com/scylladb/scylla-operator-release/pull/413